tests/end-to-end/upgrade/pkg/tests/function/function.go

```diff
@@ -0,0 +1,176 @@
+	log.Println("FunctionUpgradeTest creating resources")
+	_, err := f.createFunction(log, namespace)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (f *FunctionUpgradeTest) TestResources(stop <-chan struct{}, log logrus.FieldLogger, namespace string) error {
+	log.Println("FunctionUpgradeTest testing resources")
+	err := f.getFunctionPodStatus(namespace, 10*time.Minute)
+	if err != nil {
+		return err
+	}
+
+	host := fmt.Sprintf("http://%s.%s:8080", namespace, namespace)
+
+	value, err := f.getFunctionOutput(host, 2*time.Minute)
+	if err != nil {
+		return err
+	}
+
+	if !strings.Contains(value, f.uuid) {
+		return fmt.Errorf("Could not get expected function output:\n %v\n output:\n %v", f.uuid, value)
+	}
+
+	return nil
+}
+
+func (f *FunctionUpgradeTest) getFunctionOutput(host string, waitmax time.Duration) (string, error) {
+
+	tick := time.Tick(2 * time.Second)
+	timeout := time.After(waitmax)
+	messages := ""
+
+	for {
+		select {
+		case <-tick:
+			resp, err := http.Post(host, "text/plain", bytes.NewBufferString(f.uuid))
+			if err != nil {
+				messages += fmt.Sprintf("%+v\n", err)
+				break
+			}
+			if resp.StatusCode == http.StatusOK {
+				bodyBytes, err := ioutil.ReadAll(resp.Body)
+				if err != nil {
+					return "", err
+				}
+				return string(bodyBytes), nil
+			}
+			messages += fmt.Sprintf("%+v", err)
+
+		case <-timeout:
+			return "", fmt.Errorf("Could not get function output:\n %v", messages)
+		}
+	}
+
+}
+
+func (f *FunctionUpgradeTest) createFunction(log logrus.FieldLogger, namespace string) (*kubelessV1.Function, error) {
+	log.Println("FunctionUpgradeTest creating function")
+	function := &kubelessV1.Function{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: f.functionName,
+			Labels: map[string]string{
+				"function": f.functionName,
+			},
+		},
+		Spec: kubelessV1.FunctionSpec{
+			Handler: "handler.hello",
+			Runtime: "nodejs8",
+			Function: `module.exports = {
+				hello: function(event, context) {
+				  return(event.data)
+				}
+			  }`,
+		},
+	}
+	function, err := f.kubelessClient.KubelessV1beta1().Functions(namespace).Create(function)
+	if err != nil {
+		return nil, err
+	}
+	return function, nil
+}
+
+func (f *FunctionUpgradeTest) getFunctionPodStatus(namespace string, waitmax time.Duration) error {
+
+	timeout := time.After(waitmax)
+	tick := time.Tick(2 * time.Second)
+	for {
+		select {
+		case <-timeout:
+			pods, err := f.coreClient.CoreV1().Pods(namespace).List(metav1.ListOptions{LabelSelector: "function=" + f.functionName})
+			if err != nil {
+				return err
+			}
+			return fmt.Errorf("Pod did not start within given time  %v: %+v", waitmax, pods)
+		case <-tick:
+			pods, err := f.coreClient.CoreV1().Pods(namespace).List(metav1.ListOptions{LabelSelector: "function=" + f.functionName})
+			if err != nil {
+				return err
+			}
+			if len(pods.Items) == 0 {
+				break
+			}
+
+			if len(pods.Items) > 1 {
```

Comment:
I got an error: `Deployed 1 pod, got 2` here.
Maybe we shouldn't throw error here?